jax/lax/lax.py

```diff
@@ -1601,6 +1601,29 @@ def naryop(result_dtype, accepted_dtypes, name, translation_rule=None):
 standard_naryop = partial(naryop, _input_dtype)
 
 
+def broadcast_translate(translate):
+  def _broadcast_translate(c, *args, **kwargs):
+    # get shapes
+    arg_shapes = [c.GetShape(arg).dimensions() for arg in args]
+    kwarg_shapes = {name: c.GetShape(kwarg).dimensions() for name, kwarg in kwargs.items()}
```

Comment:
Primitives rules use keyword arguments for metadata, and they should never be XLA values. So you should ignore kwargs completely for the purpose of broadcasting and pass them through unchanged to the underlying translation rule.